### ISSUE TITLE
Improve search by ID performace

### ISSUE THREAD

**Issue by anshumanv** [anshumanv/evolutions-webapp#20]
We are iterating over each object to match it's num, this is poor in terms if performance, please improve it.

- [x] I ensure that this issue is not a duplicate of any current open issues(s).
- [ ] Assign this issue to me (I want to work on this).
- [ ] I will send my PR only after I'm assigned to this issue by any maintainer.
- [ ] I understand that I will be unassigned from this issue if I fail to send a PR for this within a week.

**Comment by deadbirdi** [comment]
What is a different way it could be done?

**Comment by Monal5031** [comment]
Well we can map the values to object and then do a BST over the numbers and retrieve back the chain we need.

**Comment by dericcain** [comment]
I gave this a shot in #46. This was a really good case for using a lookup table.
